src/EnergyPlus/PoweredInductionUnits.cc

```diff
@@ -2042,6 +2071,418 @@ void CalcParallelPIU(EnergyPlusData &state,
+        thisPIU.PriAirMassFlow = priAirMassFlowMin;
+        if (qdotDelivMinPrim >=
+            loadToHeatSetPt) { // will provide more cooling than required but not enough to drop below the heating thermostat setpoint
+            thisPIU.SecAirMassFlow = max(0.0, thisPIU.MinTotAirMassFlow - thisPIU.PriAirMassFlow);
+            thisPIU.heatingOperatingMode = HeatOpModeType::HeaterOff;
+        } else {
+            if (thisPIU.heatingControlType == HeatCntrlBehaviorType::StagedHeaterBehavior) {
+                CalcVariableSpeedPIUStagedHeatingBehavior(state, piuNum, zoneNode, loadToHeatSetPt, true, thisPIU.PriAirMassFlow);
+            } else if (thisPIU.heatingControlType == HeatCntrlBehaviorType::ModulatedHeaterBehavior) {
+                CalcVariableSpeedPIUModulatedHeatingBehavior(state, piuNum, zoneNode, loadToHeatSetPt, true, thisPIU.PriAirMassFlow);
+            }
+        }
+    } else {
+        // check how much cooling provided at max fan and primary air
+        state.dataLoopNodes->Node(thisPIU.PriAirInNode).MassFlowRate = thisPIU.MaxPriAirMassFlow;
+        Real64 TotAirMassFlow = thisPIU.MaxTotAirMassFlow;
+        state.dataLoopNodes->Node(thisPIU.SecAirInNode).MassFlowRate = max(0.0, TotAirMassFlow - thisPIU.MaxPriAirMassFlow);
+        Real64 qdotDelivMaxFan = CalcVariableSpeedPIUQdotDelivered(state, piuNum, zoneNode, false, TotAirMassFlow, 1.0);
+
+        if (zoneLoad <= qdotDelivMaxFan) { // not going to make it just run at max
+            thisPIU.PriAirMassFlow = thisPIU.PriAirMassFlow;
+            Real64 TotAirMassFlow = thisPIU.MaxTotAirMassFlow;
+            thisPIU.SecAirMassFlow = max(0.0, TotAirMassFlow - thisPIU.PriAirMassFlow);
+            thisPIU.heatingOperatingMode = HeatOpModeType::HeaterOff;
+            thisPIU.coolingOperatingMode = CoolOpModeType::CoolSecondStage;
+        } else {
+            // call regula falsi solver, vary a coooling control signal for fan speed and primary air flow together from min to max.
+            int constexpr MaxIte(500);    // Maximum number of iterations
+            Real64 constexpr Acc(0.0001); // Accuracy of result
+            int SolFla(0);                // Flag of solver
+            Real64 coolSignal = 0.5;      // starting value
+            auto f = [&state, piuNum, zoneLoad, zoneNode](Real64 const coolSignal) {
+                return CalcVariableSpeedPIUCoolingResidual(state, coolSignal, piuNum, zoneLoad, zoneNode);
+            };
+
+            General::SolveRoot(state, Acc, MaxIte, SolFla, coolSignal, f, 0.0, 1.0);
+
+            if (SolFla == -1) {
+                ShowSevereError(state, "Iteration limit exceeded in calculating variable speed fan powered box cooling signal");
+                ShowContinueErrorTimeStamp(state, "");
+                ShowFatalError(state, format("Series PIU control failed for {}:{} ", thisPIU.UnitType, thisPIU.Name));
```

Comment:
This is not good practice to fatal out on an iteration (or even SolFla ==2) error. It's better to write an error message, guess at a result, and proceed.